lightningd/lightningd.c

```diff
@@ -251,6 +253,38 @@ static void daemonize_but_keep_dir(void)
 	tal_free(cwd);
 }
 
+static void pidfile_create(const struct lightningd *ld)
+{
+        int pid_fd;
+        char *pid;
+        const tal_t *tmpctx = tal_tmpctx(NULL);
+
+        /* Create PID file */
+        pid_fd = open(ld->pidfile, O_WRONLY|O_CREAT, 0640);
+        if (pid_fd < 0)
+                errx(1, "Failed to open PID file: %s", strerror(errno));
+
+        /* Lock PID file */
+        if (lockf(pid_fd, F_TLOCK, 0) < 0)
+                /* Can't lock file */
+                errx(1, "Cannot lock PID file: %s", strerror(errno));
```

Comment:
Note that this is what they'll get if lightningd is already running, so maybe just say: `"lightningd already running?  Cannot lock PID file %s", ld->pidfile)`?